### PR TITLE
Add support for bigint

### PR DESCRIPTION
**Pull request by sigma-andex**
Add multiple deserialisation options

### CODE REVIEW

File: spago.dhall [hunk 1]
```diff
@@ -4,6 +4,8 @@
 , dependencies =
   [ "arrays"
   , "bifunctors"
+  , "bigints"
+  , "console"
```
Comment (i-am-the-slime) [inline spago.dhall:8]: Why console? Shouldn't this be in test?

Comment (sigma-andex) [reply spago.dhall:8]: good catch!

File: test/BasicsSpec.purs [hunk 2]
```diff
@@ -51,7 +52,26 @@ spec = describe "En- and decoding" $ do
       $ roundtrips (Map.fromFoldable [(Stringy "A" /\ "B"),(Stringy "C" /\ "D")])
     it "roundtrips Map with Int newtype keys"
       $ roundtrips (Map.fromFoldable [(Inty 4 /\ "B"),(Inty 8 /\ "D")])
+    it "roundtrips BigInt" do
+      let
+        inputStr = """{ "number":1, "big": 18014398509481982, "smallBig": 10 }"""
+        expected = """{"smallBig":"10","number":1,"big":"18014398509481982"}"""
```
Comment (i-am-the-slime) [inline test/BasicsSpec.purs:58]:
```suggestion
        expected = """{"smallBig":10,"number":1,"big":18014398509481982}"""
```

That's what I would *expect*

Comment (sigma-andex) [reply test/BasicsSpec.purs:58]: Unfortunately serialisation only as string

Comment (sigma-andex) [reply test/BasicsSpec.purs:58]: It's also strange that the order is reversed.

Comment (i-am-the-slime) [reply test/BasicsSpec.purs:58]: Sad this seems quite impossible without third party libraries. Can you add this to the README? That this is an additional feature to simple-json, that it requires a JS dependency and that you can't use it to produce data for APIs that expect non-string-encoded bigints?

**Review by i-am-the-slime — CHANGES_REQUESTED** [review]
Please add the capabilities and pitfalls to readme and merge this sucker.

File: test/BasicsSpec.purs (hunk 2 above)
Comment (i-am-the-slime) [reply test/BasicsSpec.purs:58]: > It's also strange that the order is reversed.

I don't care to much, it's the same JSON, I would say.

**Comment by i-am-the-slime** [comment]
Maybe we can also say that this library might help with producing bigints of the right shape: https://www.npmjs.com/package/bigint-json-native

But I'm not sure it will actually work with our modified `JSON.parse`.

**Comment by i-am-the-slime** [comment]
@sigma-andex If you fix CI, I'll merge this.

**Comment by i-am-the-slime** [comment]
Merguez

**Pull request merged**